spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/SetCurrentSnapshotProcedure.java

```diff
@@ -78,17 +83,21 @@ public StructType outputType() {
   @Override
   public InternalRow[] call(InternalRow args) {
     Identifier tableIdent = toIdentifier(args.getString(0), PARAMETERS[0].name());
-    long snapshotId = args.getLong(1);
+    Long snapshotId = args.isNullAt(1) ? null : args.getLong(1);
+    String ref = args.isNullAt(2) ? null : args.getString(2);
+    Preconditions.checkArgument(
+        snapshotId != null || ref != null, "snapshot_id and ref cannot both be null");
 
     return modifyIcebergTable(
         tableIdent,
         table -> {
           Snapshot previousSnapshot = table.currentSnapshot();
           Long previousSnapshotId = previousSnapshot != null ? previousSnapshot.snapshotId() : null;
 
-          table.manageSnapshots().setCurrentSnapshot(snapshotId).commit();
+          long sid = snapshotId != null ? snapshotId : getSnapshotIdFromRef(table, ref);
```

Comment:
Minor: We rarely use `get` in method names. What about `toSnapshotId`?
I'd also also rename `sid` into `targetSnapshotId` or something.

```
long targetSnapshotId = snapshotId != null ? snapshotId : toSnapshotId(table, ref);
table.manageSnapshots().setCurrentSnapshot(targetSnapshotId).commit();
```